api/src/main/java/uk/gov/dhsc/htbhf/claimant/reporting/ClaimantCategoryCalculator.java

```diff
@@ -0,0 +1,60 @@
+package uk.gov.dhsc.htbhf.claimant.reporting;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import uk.gov.dhsc.htbhf.claimant.entitlement.PregnancyEntitlementCalculator;
+import uk.gov.dhsc.htbhf.claimant.entity.Claimant;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.Period;
+import java.util.List;
+
+import static uk.gov.dhsc.htbhf.claimant.reporting.payload.ClaimantCategory.NOT_PREGNANT_WITH_CHILDREN;
+import static uk.gov.dhsc.htbhf.claimant.reporting.payload.ClaimantCategory.PREGNANT_AND_UNDER_16;
+import static uk.gov.dhsc.htbhf.claimant.reporting.payload.ClaimantCategory.PREGNANT_WITH_CHILDREN;
+import static uk.gov.dhsc.htbhf.claimant.reporting.payload.ClaimantCategory.PREGNANT_WITH_NO_CHILDREN;
+
+@Component
+@RequiredArgsConstructor
+public class ClaimantCategoryCalculator {
+
+    private final PregnancyEntitlementCalculator pregnancyEntitlementCalculator;
+
+    /**
+     * Determines the {@link uk.gov.dhsc.htbhf.claimant.reporting.payload.ClaimantCategory} for a given claimant and their children's date of birth.
+     *
+     * @param claimant               the claimant to determine the category of
+     * @param datesOfBirthOfChildren the dates of birth of the claimant's children
+     * @param timestamp the timestamp to use when checking a claimant/child's age
+     * @return claimant's determined category
+     */
+    public String determineClaimantCategory(Claimant claimant, List<LocalDate> datesOfBirthOfChildren, LocalDateTime timestamp) {
```

Comment:
Why does this take a LocalDateTime rather than just a LocalDate? Don't think we'd ever need the time element so would probably be better with just the LocalDate.

Reply:
This and other methods have been updated to use a LocalDate.